astropy/erfa/core.py.templ

```diff
@@ -66,7 +66,23 @@ class ErfaWarning(AstropyUserWarning):
 
 STATUS_CODES = {}  # populated below before each function that returns an int
 
+# This is a hard-coded list of status codes that need to be remapped,
+# such as to turn errors into warnings.
+STATUS_CODES_REMAP = {
+    'cal2jd': {-3: 3}
+}
+
+
 def check_errwarn(statcodes, func_name):
+    # Remap any errors into warnings in the STATUS_CODES_REMAP dict.
+    if func_name in STATUS_CODES_REMAP:
+        mapping = STATUS_CODES_REMAP[func_name]
+        for i in range(len(statcodes)):
```

Comment:
Also thought, why not just
```
statcodes[statcodes == before] = after
```
But clearly one should test, not just think things are faster:
```
a = np.arange(10000.)
def replace(a, before, after):
    mask = a == before
    if np.any(mask):
        a[mask] = after
    return a

%timeit replace(a, -1, 0)
# 10000 loops, best of 3: 17.4 µs per loop
%timeit np.where(a == -1, 0, a)
# 10000 loops, best of 3: 128 µs per loop
%timeit a[a == -1] = 0
# 100000 loops, best of 3: 14 µs per loop
```
So, the winner seems to be the simplest one!